### ISSUE TITLE
Spoon run complains about other non-related project 

### ISSUE THREAD

**Issue by imbeyondboredom** [square/spoon#284]
When I run
`./gradlew clean :app:spoonDevelopmentAndroidTest`
It responds with 

```
2015-09-05 15:37:59 [SR.runTests] Executing instrumentation suite on 1 device(s).
2015-09-05 15:38:17 [SDR.run] Unable to find test for com.other.ui.HomeTest#takeScreenshots
2015-09-05 15:38:17 [SDR.run] Unable to find test for com.other.ui.HomeTest#takeScreenshots
2015-09-05 15:38:17 [SDR.run] Unable to find test for com.other.ui.HomeTest#takeScreenshots
2015-09-05 15:38:17 [SDR.run] Unable to find test for com.other.ui.HomeTest#takeScreenshots
2015-09-05 15:38:17 [SDR.run] Unable to find test for com.other.ui.HomeTest#takeScreenshots

BUILD SUCCESSFUL
```

The thing is that com.other.ui.HomeTest is a test in a completely different project in a different folder that also uses spoon. If I grep for com.other I get this:

```
 grep -r com.other .
Binary file ./.gradle/2.4/taskArtifacts/fileSnapshots.bin matches
```

I can delete that file and run the tests again but it just generates it again.

if I grep for com.other in ~/.gradle/ I see a ton of files (mostly logs). I deleted all of ~/.gradle/cache in case something was weird there but it made no difference.

This happens on at least 1.1.10 and 1.2.0

I'll also say that I can't get spoon to run my test that I want to run. Not sure if it's related to this or not but it's hard to debug when it says it's looking for the wrong project.

Here is my gradle file:

```
apply plugin: 'com.android.application'
apply plugin: 'spoon'

android {
    compileSdkVersion 23
    buildToolsVersion "23.0.0"

    defaultConfig {
        applicationId "com.willowtreeapps.testmeout"
        minSdkVersion 16
        targetSdkVersion 23
        versionCode 1
        versionName "1.0"
    }
    buildTypes {
        development {
            signingConfig signingConfigs.debug
            debuggable true
        }
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }
    }
    testOptions {
        unitTests.returnDefaultValues = true
    }


    lintOptions {
        abortOnError false
    }

    testBuildType 'development'
}

dependencies {
    compile fileTree(include: ['*.jar'], dir: 'libs')
    compile 'com.android.support:appcompat-v7:23.0.0'
    compile 'com.android.support:support-annotations:23.0.0'
    testCompile 'junit:junit:4.12'
    developmentCompile 'com.squareup.spoon:spoon-client:1.2.0'
    developmentCompile 'com.android.support:support-annotations:23.0.0'
    developmentCompile 'com.android.support.test:runner:0.3'
    developmentCompile 'com.android.support.test:rules:0.3'
    developmentCompile 'com.android.support.test.espresso:espresso-core:2.2'
//    androidTestCompile 'com.squareup.spoon:spoon-client:1.1.10'
//    androidTestCompile 'com.android.support:support-annotations:23.0.0'
//    androidTestCompile 'com.android.support.test:runner:0.3'
//    androidTestCompile 'com.android.support.test:rules:0.3'
//    androidTestCompile 'com.android.support.test.espresso:espresso-core:2.2'
}
```

**Comment by gdaniels** [comment]
Just tracked this down myself - what's happening here is that Spoon is trying to match up the screenshots in the screenshot dir with the tests that have been run.  The relevant code is in [SpoonDeviceRunner.java](https://github.com/square/spoon/blob/master/spoon-runner/src/main/java/com/squareup/spoon/SpoonDeviceRunner.java#L291).  If screenshots have been left around by previous runs for other apps, then you'll see this error.  I'm not yet sure why Spoon isn't clearing the screenshot dir at the beginning of a run, but you can get rid of the error by clearing it out manually (blow away /sdcard/app_spoon-screenshots).

**Comment by bagintz** [comment]
me too

**Comment by dweebo** [comment]
This should be fixed by #482, though issue will still be there for any saved files across test runs
